cmd/resim/commands/batch.go

```diff
@@ -346,6 +351,93 @@ func createBatch(ccmd *cobra.Command, args []string) {
+		SuiteUrl   string
+		SuiteName  string
+		BatchUrl   string
+		SystemUrl  string
+		SystemName string
+	}{
+		fmt.Sprintf("%s/test-suites/%s/revisions/%d", baseUrl, batch.TestSuiteID.String(), *batch.TestSuiteRevision),
+		suite.Name,
+		fmt.Sprintf("%s/batches/%s", baseUrl, batch.BatchID.String()),
+		fmt.Sprintf("%s/systems/%s", baseUrl, batch.SystemID.String()),
+		system.Name,
+	}
+	introTemplate := template.Must(template.New("intro").Parse("Last night’s <{{.SuiteUrl}}|{{.SuiteName}}> *<{{.BatchUrl}}|run>* for <{{.SystemUrl}}|{{.SystemName}}> ran successfully with the following breakdown:"))
+	var introBuffer bytes.Buffer
+	introTemplate.Execute(&introBuffer, introData)
+	introTextBlock := slack.NewTextBlockObject("mrkdwn", introBuffer.String(), false, false)
+	blocks.BlockSet[0] = slack.NewSectionBlock(introTextBlock, nil, nil)
+
+	// List section
+	boldStyle := slack.RichTextSectionTextStyle{Bold: true}
+	buildListElement := func(count int, label string, filter string) *slack.RichTextSection {
+		return slack.NewRichTextSection(
+			slack.NewRichTextSectionTextElement(fmt.Sprintf("%d ", count), nil),
+			slack.NewRichTextSectionLinkElement(introData.BatchUrl+"?performanceFilter="+filter, label, &boldStyle),
+		)
+	}
+
+	listBlock := slack.NewRichTextList("bullet", 0,
+		slack.NewRichTextSection(slack.NewRichTextSectionTextElement(fmt.Sprintf("%d total tests", *batch.TotalJobs), nil)),
+		// Passed calculation borrowed from bff: https://github.com/resim-ai/rerun/blob/ebf0cde9472f555ae099e08e512ed4a7dfdf01f4/bff/lib/bff/batches/conflated_status_counts.ex#L49
+		buildListElement(batch.JobStatusCounts.Succeeded-(batch.JobMetricsStatusCounts.FailBlock+batch.JobMetricsStatusCounts.FailWarn), "Passed", "Passed"),
+		buildListElement(batch.JobMetricsStatusCounts.FailBlock, "Blocking", "Blocker"),
+		buildListElement(batch.JobMetricsStatusCounts.FailWarn, "Warning", "Warning"),
+		buildListElement(batch.JobStatusCounts.Error, "Erroring", "Error"),
+	)
+	blocks.BlockSet[1] = slack.NewRichTextBlock("list", listBlock)
+
+	// Report section
+	reportResponse, err := Client.ListReportsWithResponse(context.Background(), *batch.ProjectID, &api.ListReportsParams{Search: Ptr(fmt.Sprintf("test_suite_id = \"%s\"", batch.TestSuiteID.String())), OrderBy: Ptr("timestamp")})
+	if err != nil {
+		log.Fatal("unable to retrieve report for batch:", err)
+	}
+	ValidateResponse(http.StatusOK, "unable to retrieve report for batch", reportResponse.HTTPResponse, reportResponse.Body)
+	reports = reportResponse.JSON200.Reports
+	if len(*reports) > 0 {
+		reportFooter := fmt.Sprintf("See more historical details in the nightly *<%s/reports/%s|Report>*.", baseUrl, (*reports)[0].ReportID)
+		reportTextBlock := slack.NewTextBlockObject("mrkdwn", reportFooter, false, false)
+		blocks.BlockSet[2] = slack.NewSectionBlock(reportTextBlock, nil, nil)
+	} else {
+		blocks.BlockSet = blocks.BlockSet[:2]
+	}
+
+	webhookPayload := slack.WebhookMessage{
+		Username: "ReSim",
```

Comment:
this is the username shown in Slack? Does that need to match anything with their slack bot?

Reply:
It is. I'll just let them configure it instead...